ProcessLib/BoundaryCondition/GenericNaturalBoundaryCondition-impl.h

```diff
@@ -20,17 +19,11 @@ template <typename Data>
 GenericNaturalBoundaryCondition<BoundaryConditionData,
                                 LocalAssemblerImplementation>::
     GenericNaturalBoundaryCondition(
-        typename std::enable_if<
-            std::is_same<typename std::decay<BoundaryConditionData>::type,
-                         typename std::decay<Data>::type>::value,
-            bool>::type is_axially_symmetric,
         unsigned const integration_order, unsigned const shapefunction_order,
         NumLib::LocalToGlobalIndexMap const& dof_table_bulk,
         int const variable_id, int const component_id,
         unsigned const global_dim, MeshLib::Mesh const& bc_mesh, Data&& data)
-    : _data(std::forward<Data>(data)),
-      _bc_mesh(bc_mesh),
-      _integration_order(integration_order)
+    : _data(std::forward<Data>(data)), _bc_mesh(bc_mesh)
```

Comment:
?

Reply:
In other words: The construct
```
typename std::enable_if<
            std::is_same<typename std::decay<BoundaryConditionData>::type,
                         typename std::decay<Data>::type>::value,
            bool>::type
```
was there on purpose. Since this templated constructor might lead to nasty error messages, some nicer hints should be provided. E.g., the following (better code) has been used in some later code:
```
    static_assert(std::is_same<typename std::decay<BoundaryConditionData>::type,
                               typename std::decay<Data>::type>::value,
"Type mismatch between declared and passed BC data.");
```